### PR TITLE
typescript + fixes

### PR DESCRIPTION
**Pull request by alexdresko**
I totally just picked this random javascript library and converted it to typescript. 

In this particular case, there wasn't a lot to gain, except that a type definition is generated. The type definition make it much easier to utilize this library in a typescript project. 

VS Code, of course, provides the best editing experience, so you'll see a few VS Code specific project files.

Don't know if the original author will even care for this PR. I just wanted to do something interesting this Friday morning.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,9 +33,12 @@
   "homepage": "https://github.com/arshadkazmi42/custom-date-formatting#readme",
   "devDependencies": {
     "chai": "^4.1.1",
-    "mocha": "^3.5.0"
+    "jshint": "^2.9.6",
+    "mocha": "^3.5.0",
+    "typescript": "^3.2.1"
   },
   "dependencies": {
+    "@types/node": "^10.12.10",
```
Comment (arshadkazmi42) [inline package.json]: Why this has been added?

**Comment by alexdresko** [comment]
@arshadkazmi42 I made all of the requested changes, except for adding tests. I don't know enough about the library to know how to test it. I only know typescript :)  Can you pull my branch and make sure it works in your environment?

**Comment by arshadkazmi42** [comment]
@alexdresko everything looks good. I checked the existing code is fine and tests are passing. 
Can you follow the tests in `test/index.js` and write something similar for the `.ts` version of the code also?

File: package.json (hunk 1 above)
Comment (alexdresko) [reply package.json]: Because `require(...)` isn't understood otherwise. It's not a native js function, to my knowledge.

**Comment by alexdresko** [comment]
@arshadkazmi42 Doh! I totally forgot about the test file! I'll take a stab at it.

**Comment by alexdresko** [comment]
@arshadkazmi42 I've converted the test file to ts. Had to tweak a couple things along the way, as you'll see. Seems to be correct, though.

File: package.json [hunk 2]
```diff
@@ -30,9 +33,13 @@
...
   "devDependencies": {
     "chai": "^4.1.1",
-    "mocha": "^3.5.0"
+    "jshint": "^2.9.6",
+    "mocha": "^3.5.0",
+    "typescript": "^3.2.1"
   },
   "dependencies": {
+    "@types/chai": "^4.1.7",
+    "@types/jest": "^23.3.10",
```
Comment (arshadkazmi42) [inline package.json]: Do we need these two dependencies here? If its only for the tests, we can move these two dependencies to `devDependencies`

**Comment by alexdresko** [comment]
I did not create any tests. I only converted the existing tests to ts. I'll try running the tests and see what happens.

File: package.json (hunk 2 above)
Comment (alexdresko) [reply package.json]: Done.

File: test/index.js [hunk 3]
```diff
@@ -1,39 +1,30 @@
-var moment = require("moment");
-var should = require('chai').should(),
```
Comment (arshadkazmi42) [inline test/index.js]: You have removed this import, due to this tests are failing. Can. you add that `should` import to the test.

Comment (alexdresko) [reply test/index.js]: Ahhhh that did it! I had to add the node typings back.  

I'm still learning the intricacies amongst `import x = require('y')`, `import x from 'y'`, and `var x = require('y')`.  One of the reasons I'm trying to help convert js projects to ts is to learn more about that kind of stuff. 

Anything else?

**Comment by alexdresko** [comment]
`tsc` will compile the ts files to js. Then I just ran `mocha .\test\index.js` as you did. I don't know if mocha can run ts files directly.

**Comment by arshadkazmi42** [comment]
I found this command to test the `ts` file 
`mocha -r ts-node/register test/index.ts`

This looks awesome, thank you for all the work and changes. I am getting this merged. 🎉

**Comment by alexdresko** [comment]
Very glad you like it! Hope typescript helps you do even more awesome stuff with this library.

**Pull request merged**